### DISCUSSION TITLE
plugin request: sitemap.xml (👍10)

### DISCUSSION THREAD

**Discussion by zachleat** [11ty/eleventy#4014] [Enhancement Queue]
Related 11ty/eleventy#394, 11ty/eleventy#417 

See https://github.com/jekyll/jekyll-sitemap

**Comment by Ryuno-Ki** [comment]
For starters, this is my current implementation:

```njk
---
permalink: /sitemap.xml
excludeFromSitemap: true
---
<?xml version='1.0' encoding='UTF-8'?>
<urlset xmlns='http://www.sitemaps.org/schemas/sitemap/0.9'>
  {%- for item in collections.all %}
    {%- if not item.data.excludeFromSitemap %}
    <url>
      <loc>https://jaenis.ch{{ item.url }}</loc>
      <lastmod>{{ item.date | isoDate }}</lastmod>
      <changefreq>{{ item.data.changefreq }}</changefreq>
      <priority>{{ item.data.priority }}</priority>
    </url>
    {%- endif %}
  {%- endfor %}
</urlset> 
```

seeAlso: https://github.com/11ty/11ty-website/issues/1853 + https://github.com/11ty/eleventy-base-blog/pull/22/files

**Comment by edwardhorsford** [comment]
And here's mine. Note I've got a filter that removes items without permalinks (which I'm using until there's first class support for drafts).


``` nunjucks
---
permalink: /sitemap.xml
sitemapIgnore: true
---
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
{%- for item in collections.all | permalinkNotFalse | sort(false, false, 'url') %}
{%- if item.data.sitemapIgnore !== true %}
  <url>
    <loc>https://www.edwardhorsford.com{{ item.url }}</loc>
    <lastmod>{{ item.date | formatDate("dashDate") }}</lastmod>
    <changefreq>{{ item.data.changeFrequency or globals.sitemap.defaultChangeFrequency }}</changefreq>
    <priority>{{ item.data.sitemapPriority or globals.sitemap.defaultPriority }}</priority>
  </url>
{% endif %}
{% endfor %}
</urlset>
```

**Comment by Ryuno-Ki** [comment]
Okay, so how could a plugin look like?
I'd say, a shortcode, which accepts the „globals” Data file content as argument?
It would depend on some data attributes for ignoring then. Plus, I can see a dependency to a date function (I'm about to rewrite my filter to use [date-fns](https://www.npmjs.com/package/date-fns)).

Am I missing something?

**Comment by brycewray** [comment]
Any thoughts on something that would (also?) work with [.11ty.js templating](https://www.11ty.dev/docs/languages/javascript/) and/or [universal filters](https://www.11ty.dev/docs/filters/#universal-filters)? Since [eleventy-rss-helper](https://www.npmjs.com/package/eleventy-rss-helper) seems to be how one would do an RSS feed using .11ty.js, sitemaps may be the last frontier for going all-.11ty.js. 😄

**Comment by TheFrugalDeveloper** [comment]
I have a basic implementation of a sitemap file using `.11ty.js` syntax.

Find it at my WIP boilerplate: [Eleventy-Core: Sitemap Implementation](https://github.com/TheFrugalDeveloper/eleventy-core/blob/master/_src/sitemap.11ty.js) 

It relies on computed data quite a bit, and it's not fully fleshed out, but it's a start.
